salt/_pillar/azureblob.py

```diff
@@ -0,0 +1,449 @@
+# -*- coding: utf-8 -*-
+'''
+Use Azure Blob as a Pillar source
+
+.. versionadded:: Sodium
+
+:maintainer: <devops@eitr.tech>
+:maturity: new
+:depends:
+:depends:
+    * `azure-storage-blob <https://pypi.org/project/azure-storage-blob/>`_ >= 12.0.0
+
+The Azure Blob external pillar can be configured with the following parameters:
+
+.. code-block:: yaml
+
+    ext_pillar:
+      - azureblob:
+          container:
+          connection_string:
+          multiple_env:
+          environment:
+          blob_cache_expire:
+          blob_sync_on_update:
+
+:param container: The name of the target Azure Blob Container.
+
+:param connection_string: The connection string to use to access the specified Azure Blob Container.
+
+:param multiple_env: Specifies whether the pillar should interpret top level folders as pillar environments.
+    Defaults to false.
+
+:param environment: Specifies which environment the container represents when in single environment mode. Defaults
+    to 'base' and is ignored if multiple_env is set as True.
+
+:param blob_cache_expire: Specifies expiration time of the Azure Blob metadata cache file. Defaults to 30s.
+
+:param blob_sync_on_update: Specifies if the cache is synced on update. Defaults to True.
+
+'''
+# Import Python libs
+from __future__ import absolute_import, print_function, unicode_literals
+import logging
+import os
+import time
+import pickle
+from copy import deepcopy
+
+# Import 3rd-party libs
+# pylint: disable=import-error,no-name-in-module,redefined-builtin
+from salt.ext import six
+from salt.ext.six.moves import filter
+# pylint: enable=import-error,no-name-in-module,redefined-builtin
+
+# Import Azure libs
+HAS_LIBS = False
+try:
+    from azure.storage.blob import BlobServiceClient
+    HAS_LIBS = True
+except ImportError:
+    pass
+
+# Import Salt libs
+from salt.pillar import Pillar
+import salt.utils.files
+import salt.utils.hashutils
+
+__virtualname__ = 'azureblob'
+
+# Set up logging
+log = logging.getLogger(__name__)
+
+
+def ext_pillar(minion_id,
+               pillar,  # pylint: disable=W0613
+               container,
+               connection_string,
+               multiple_env=False,
+               environment='base',
+               prefix='',
```

Comment:
We discussed `prefix` briefly. I think it was intended to namespace pillar data in an S3 bucket. A bucket ~= a blob, so the namespacing in a blob is by container. So, I think we can drop the prefix parameter here.